components/base/wheeled/wheeled_base.go

```diff
@@ -130,64 +130,56 @@ func (wb *wheeledBase) Reconfigure(ctx context.Context, deps resource.Dependenci
 
 	wb.spinSlipFactor = newConf.SpinSlipFactor
 
-	// Check if wb.left is different from newConf.Left before changing wb.left
-	if len(wb.left) != len(newConf.Left) {
-		// Resetting the left motor list
-		wb.left = make([]motor.Motor, 0)
-
-		for _, name := range newConf.Left {
-			m, err := motor.FromDependencies(deps, name)
-			if err != nil {
-				return errors.Wrapf(err, "no left motor named (%s)", name)
+	updateMotors := func(curr *[]motor.Motor, fromConfig *[]string) ([]motor.Motor, error) {
```

Comment:
Linter forced me to factor out this code into a helper function because it's duplicative between left and right.

Reply:
should we add someone from components team to review this bit?